soroban-env-host/src/host/conversion.rs

```diff
@@ -150,4 +155,87 @@ impl Host {
+                self.map_err(data.try_into())?,
+            ))),
+        }
+    }
+
+    fn event_topic_from_rawval(&self, topic: RawVal) -> Result<ScVal, HostError> {
+        self.charge_budget(CostType::ValXdrConv, 1)?;
+        if topic.is_u63() {
+            Ok(ScVal::U63(unsafe { topic.unchecked_as_u63() }))
+        } else {
+            match topic.get_tag() {
+                Tag::Object => {
+                    unsafe {
+                        self.unchecked_visit_val_obj(topic, |ob| {
+                            // charge budget
+                            let sco = match ob {
+                                None => Err(self.err_status(ScHostObjErrorCode::UnknownReference)),
+                                Some(ho) => match ho {
+                                    // TODO: use more event-specific error codes than `UnexpectedType`
+                                    HostObject::Vec(_) => {
+                                        Err(self.err_status(ScHostObjErrorCode::UnexpectedType))
+                                    }
+                                    HostObject::Map(_) => {
+                                        Err(self.err_status(ScHostObjErrorCode::UnexpectedType))
+                                    }
+                                    HostObject::Bin(b) => {
+                                        if b.len() > 32 {
+                                            return Err(
+                                                self.err_status(ScHostObjErrorCode::UnexpectedType)
+                                            );
+                                        }
+                                        Ok(ScObject::Bytes(self.map_err(b.clone().try_into())?))
+                                    }
+                                    HostObject::U64(u) => Ok(ScObject::U64(*u)),
+                                    HostObject::I64(i) => Ok(ScObject::I64(*i)),
+                                    HostObject::BigInt(bi) => self.scobj_from_bigint(bi),
+                                    HostObject::Hash(_) => todo!(),
+                                    HostObject::PublicKey(_) => todo!(),
+                                },
+                            }?;
+                            Ok(ScVal::Object(Some(sco)))
+                        })
+                    }
+                }
+                _ => self.from_host_val(topic),
+            }
+        }
+    }
+
+    pub(crate) fn event_topics_from_host_obj(&self, topics: Object) -> Result<ScVec, HostError> {
+        unsafe {
+            self.unchecked_visit_val_obj(topics.into(), |ob| {
+                self.charge_budget(CostType::ValXdrConv, 1)?;
+                match ob {
+                    None => Err(self.err_status(ScHostObjErrorCode::UnknownReference)),
+                    Some(ho) => match ho {
+                        HostObject::Vec(vv) => {
+                            if vv.len() > 4 {
```

Comment:
💡 Same here, this value would benefit from being in a constant.

Reply:
Fixed.